### PR TITLE
Merge dev into master for v0.29 release

### PR DESCRIPTION
**Pull request by albi3ro**
Merge the dev branch into master after the v0.29 release.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -16,13 +16,12 @@ pyscf==2.1.1
...
 openfermionpyscf==0.5   # required by 2 tutorials: tutorial_quantum_chemistry and tutorial_adaptive_circuits and tutorial_diffable_shadows
-pennylane-lightning
-pennylane==0.28.0
-pennylane-sf==0.20.1
-pennylane-cirq==0.28.0
-pennylane-qiskit==0.28.0
+git+https://github.com/PennyLaneAI/pennylane.git
+git+https://github.com/PennyLaneAI/pennylane-sf.git
+git+https://github.com/PennyLaneAI/pennylane-cirq.git
+git+https://github.com/PennyLaneAI/pennylane-qiskit.git
```
Comment (trbromley) [inline requirements.txt:22]: Does this need to be updated to a pinned version?

**Review by trbromley — COMMENTED** [review]
Thanks @albi3ro!

File: requirements.txt [hunk 2]
```diff
@@ -16,13 +16,12 @@ pyscf==2.1.1
 cirq-core==0.14.1
 cirq-pasqal==0.14.1
 openfermionpyscf==0.5   # required by 2 tutorials: tutorial_quantum_chemistry and tutorial_adaptive_circuits and tutorial_diffable_shadows
-pennylane-lightning
-pennylane==0.28.0
+pennylane==0.29.0
 pennylane-sf==0.20.1
-pennylane-cirq==0.28.0
-pennylane-qiskit==0.28.0
+pennylane-cirq==0.29.0
```
Comment (trbromley) [inline requirements.txt:21]: Did we release a new `pennylane-cirq`? I don't see it on PyPI

**Review by trbromley — APPROVED** [review]

File: requirements.txt (hunk 2 above)
Comment (albi3ro) [reply requirements.txt:21]: about to be updated. there were a couple hiccups.

Comment (albi3ro) [reply requirements.txt:21]: and now is on pypi.

**Review by mlxd — APPROVED** [review]

**Pull request merged**